synapse/http/client.py

```diff
@@ -111,20 +128,32 @@ def _callback(addrs):
 
 
 class BlacklistingAgentWrapper(Agent):
-    def __init__(self, agent, reactor, whitelist=None, blacklist=None):
+    """
+    An Agent wrapper which will prevent access to IP addresses being accessed
+    directly (without an IP address lookup).
```

Comment:
I'm not really clear what you mean by "an IP address lookup" here.

Is this just an optimisation? I'm not sure the extra complexity it introduces makes it worthwhile if so.

Reply:
Accessing an IP address directly means that it doesn't do a DNS lookup for said IP address.

Reply:
right, so it never hits the IPBlacklistingResolver? Is this not a problem if we get a 302 redirect to an IP address?